pkg/partials/partial.go

```diff
@@ -3,29 +3,50 @@ package partials
 import (
 	"encoding/json"
 	"log"
+
+	"github.com/pokt-network/poktroll/pkg/partials/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
 )
 
+// GetRequestType returns the request type for the given payload.
+func GetRequestType(payloadBz []byte) (sharedtypes.RPCType, error) {
+	partialRequest, err := partiallyUnmarshalRequest(payloadBz)
+	if err != nil {
+		return sharedtypes.RPCType_UNKNOWN_RPC, err
+	}
+	return partialRequest.GetRPCType(), nil
+}
+
 // GetErrorReply returns an error reply for the given payload and error,
 // in the correct format required by the request type.
 func GetErrorReply(payloadBz []byte, err error) ([]byte, error) {
-	log.Printf("DEBUG: Partially Unmarshalling request: %s", string(payloadBz))
 	partialRequest, er := partiallyUnmarshalRequest(payloadBz)
 	if er != nil {
 		return nil, er
 	}
 	return partialRequest.GenerateErrorPayload(err)
 }
 
+// GetMethodWeighting returns the weighting for the given request.
+func GetMethodWeighting(payloadBz []byte) (uint64, error) {
+	partialRequest, err := partiallyUnmarshalRequest(payloadBz)
+	if err != nil {
+		return 0, err
+	}
+	return partialRequest.GetMethodWeighting()
+}
+
 // partiallyUnmarshalRequest unmarshals the payload into a partial request
 // that contains only the fields necessary to generate an error response and
 // handle accounting for the request's method.
 func partiallyUnmarshalRequest(payloadBz []byte) (PartialPayload, error) {
```

Comment:
Curious to know how the RPCType is inferred when we'll have more than one

Reply:
Here will be sort of like a switch logic but instead but unmarshalling - the first example would fail if it was not a json request with at least one of those fields. If this is not the case we will get to the next type - once i have a better understanding of REST request format I will put it in it will make more sense how we add RPC types